packages/app/obojobo-express/server/routes/api/drafts.js

```diff
@@ -102,8 +106,20 @@ router
 router
 	.route('/new')
 	.post(requireCanCreateDrafts)
-	.post((req, res, next) => {
+	.post((req, res) => {
 		return DraftModel.createWithContent(req.currentUser.id, draftTemplate, draftTemplateXML)
+			.then(async newDraft => {
+				if (req.body.collectionId) {
+					const hasPerms = await userHasPermissionToCollection(
+						req.currentUser.id,
+						req.body.collectionId
+					)
+					if (hasPerms) {
+						CollectionModel.addModule(req.body.collectionId, newDraft.id, req.currentUser.id)
+					}
+				}
+				return newDraft
+			})
```

Comment:
Good point. I think I finalized this when the idea was still that the 'collections/:collectionId' page would be public but otherwise all the tools would be available based on permissions - so it'd theoretically be possible for a person to be viewing a collection they have no perms to when they try making a new module. I didn't go back and adjust this after making private collections more restricted.

Will adjust to throw an error in the case of somebody trying to add a module to a collection they don't have perms to.